frontend/connection/src/DoInitPings.tsx

```diff
@@ -153,9 +204,38 @@ If you are trying to access a Streamlit app running on another server, this coul
           // The request was made but no response was received
           // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
           // http.ClientRequest in node.js
+
+          if (shouldSendClientError) {
+            // Handle retrieving the source URL from the error (health or host-config endpoint)
+            const source = error.request.path
+              ? new URL(error.request.path).pathname
+              : "DoInitPings"
+            LOG.error(
+              `Client Error in reaching server endpoint - No response received when attempting to reach ${source}`
+            )
+            sendClientError(
+              "No response received from server",
+              error.request.status,
+              source
+            )
+          }
           return retryWhenTheresNoResponse()
         }
         // Something happened in setting up the request that triggered an Error
+        if (shouldSendClientError) {
+          // Handle retrieving the source URL from the error (health or host-config endpoint)
+          const source = error.config?.url
+            ? new URL(error.config.url).pathname
```

Comment:
I assume the url is in fact a URL, but same question about parsing. Should we have a convenience function or something.

```js
function determineUrlSource(url) {
  let source = "DoInitPings"
  
  if (url) {
    try {
      source = new URL(url).pathname
    } catch(e) {
      LOG.error(`unrecognized url: ${url}`)
    }
  }

  return source;
}